src/clj/conceptual/core.clj

```diff
@@ -433,54 +433,61 @@
       ~@bodies
       (apply-aggregator! ~(first binding)))))
 
-;; TODO: support more compaction target types
+
+(defn- db-type
+  [db]
+  (let [db-type (type db)]
+    (if (instance? Class db-type)
+      (.getName ^Class db-type)
+      db-type)))
+
+(defmulti compact-db! (fn [db _type] [(db-type db) _type]))
```

Comment:
wonder if there should be a java interface for compact, pickle, and unpickle?
With the multimethod approach would need a defmethod for each subtype of a DB class.  Protocol would be another option.

Reply:
multimethod would be better I feel in this case